docs/design/ConstructiveSolidGeometry.MD

```diff
@@ -0,0 +1,136 @@
+# Constructive Solid Geometry
+
+[Constructive Solid Geometry](https://en.wikipedia.org/wiki/Constructive_solid_geometry) (CSG) allows the modeler to create
+complex shapes using boolean operators to combine simpler
+solid objects.
+
+![](https://upload.wikimedia.org/wikipedia/commons/thumb/8/8b/Csg_tree.png/450px-Csg_tree.png)
+
+In Shards we employ a [Signed Distance Function](https://en.wikipedia.org/wiki/Signed_distance_function) (SDF) to generate implicit 3D shapes such as spheres, cubes, capsules, torii etc. which can be traced by ray-marching through the scene.
+
+We'll explore in depth the binary tree representation, how to flatten it to a list, how the shader consumes it and what's the math behind it.
```

Comment:
Add a comma/ rewrite
`consumes it and what's the math behind it` => `consumes it, and the math behind it all`